### DISCUSSION TITLE
Bind Mouse Buttons

### DISCUSSION THREAD

**Discussion by rrveex** [YaLTeR/niri#486] [Ideas]
Ability to configure mouse buttons (at least side "back" and "forward") actions.

For instance, in hyprland I have:

`bind = , mouse:275, exec, wtype -M ctrl -k w -m ctrl     # mouse "back" -> ctrl+w (close tab)`

**Comment by rrveex** [comment]
I wrote a draft here:

https://github.com/YaLTeR/niri/compare/main...rrveex:niri:mouse_extra?expand=1

This simplistic approach doesn't take into consideration different mice - mine for instance sends different codes than the backend library assumes. So I hard-coded mine :)

Also, if back/fw. are not mapped in config, I ignore them (on purpose).

Something like `mouse:event_code` in config would be of course nicer....

**Reply by YaLTeR** [reply]
Hm, what's with the different codes? I thought they should be standardized

**Reply by rrveex** [reply]
I couldn't find documentation. Anyway, in smithay's definition, there's a gap between 1,2,3 and back/fw - and my side buttons are exactly in that gap.

**Reply by YaLTeR** [reply]
Maybe just a smithay error then

**Reply by rrveex** [reply]
> Maybe just a smithay error then

Not necessarily an error. Might be the manufacturer of his mouse relied on forced-installed malware "drivers" for windows so they didn't care what's sent via usb.

Regardless, mice with more than 5 buttons are not that unusual; I don't care because I rarely use the mouse (actually only in the browser, and "close tab" is enough for me), but perhaps others...

Your decision, please tell me if you want me to rewrite and do a PR or you're just going to implement it some day (which I'm sure would look nicer, since I'm no big rust fan :) )

**Reply by YaLTeR** [reply]
Feel free to do as you feel like; I'll probably get to it at some point down the line anyway

**Reply by rrveex** [reply]
Since noone else seems to be interested since I opened the topic and I have my patch, I'll leave it in your competent hands :)

Btw, *very* nice job with `niri`, thank you!

**Comment by fidelix** [comment]
Any progress on this in some other front, maybe?

**Reply by fidelix** [reply]
Never mind, apparently this was solved here:
https://github.com/YaLTeR/niri/pull/865

I suppose this discussion should be closed.

**Reply by YaLTeR** [reply]
Yes

I don't really check and close discussions, only issues
